include/aws/crt/cbor/Cbor.h

```diff
@@ -0,0 +1,367 @@
+
+                /**
+                 * Get the next ArrayStart element. Only consume the ArrayStart element and set the size of array to
+                 * &out_size, not the content of the array. The next &out_size CBOR data items will be the content of
+                 * the array for a valid CBOR data.
+                 *
+                 * Notes: For indefinite-length, this function will fail with "AWS_ERROR_CBOR_UNEXPECTED_TYPE". The
+                 * designed way to handle indefinite-length is:
+                 * - Get IndefArrayStart from PeekType
+                 * - Call ConsumeNextSingleElement to pop the indefinite-length start.
+                 * - Decode the next data item until Break is read.
+                 *
+                 * @return If successful, return the size of array
+                 *         If not, return will be none and Aws::Crt::LastError() can be
+                 *          used to retrieve CRT error code.
+                 */
+                Optional<uint64_t> PopNextArrayStart() noexcept;
+
+                /**
+                 * Get the next MapStart element. Only consume the MapStart element and set the size of array to
+                 * &out_size, not the content of the map. The next &out_size pair of CBOR data items as key and value
+                 * will be the content of the array for a valid CBOR data.
+                 *
+                 * Notes: For indefinite-length, this function will fail with "AWS_ERROR_CBOR_UNEXPECTED_TYPE". The
+                 * designed way to handle indefinite-length is:
+                 * - Get IndefMapStart from PeekType
+                 * - Call ConsumeNextSingleElement to pop the indefinite-length start.
+                 * - Decode the next data item until Break is read.
+                 *
+                 * @return If successful, return the size of map
+                 *         If not, return will be none and Aws::Crt::LastError() can be
+                 *          used to retrieve CRT error code.
+                 */
+                Optional<uint64_t> PopNextMapStart() noexcept;
+
+                /**
+                 * Get the next Tag element. Only consume the Tag element and set the tag value to out_tag_val,
+                 * not the content of the tagged value. The next CBOR data item will be the content of the tagged value
+                 * for a valid CBOR data.
+                 *
+                 * @return If successful, return the tag value
+                 *         If not, return will be none and Aws::Crt::LastError() can be
+                 *          used to retrieve CRT error code.
+                 */
+                Optional<uint64_t> PopNextTagVal() noexcept;
+
+                /**
+                 * @return the value of the last aws error encountered by operations on this instance.
+                 */
+                int LastError() const noexcept { return m_lastError ? m_lastError : AWS_ERROR_UNKNOWN; }
```

Comment:
You have this function, but wrote all the other docs to say "If not, return will be none and Aws::Crt::LastError() can be used to retrieve CRT error code."

Alter docs to tell users that they should use it, or just delete this (and m_lastError)